lib/editable-project-domain.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { OptimisticTextInput } from './optimistic-inputs';
+import { CodeExample, PropsDefinition, Prop } from './story-utils';
+import { Button } from './button';
+
+/*
+  Editable Project Domain (aka project name):
+  - does not control permissions/authorization
+  - has internal state:
+    - this lets a user type out invalid and in-progress names without necessarily updating the page state
+  - expects to be within external state that can be updated with onChange and onBlur:
+    - onChange should be used to persist changes everytime the user types (also potentially updating the url in the browser so as to more gracefully handle refreshes)
+    - onBlur should be used to update state globally (no need to cause global rerenders until we're ready)
+  - as with all optimisticTextInputs if an error is hit in the onchange or onblur, it'll show a default error state (right now that's a little firetruck)
```

Comment:
We should add this part about errors to the Story - usually when I (and presumably others) are looking for docs, I go straight to the glitch.me/rendered stories page and not the code unless I really need to dig deep.

Reply:
it's in the story, I made a button to fake an error, is there more that you would want to see?

Reply:
Ah just the comment, in words, that that error that you can fake was surfaced through the onchange or onblur functions and not some other method (like an onError thing like I was initially thinking)